arches/app/models/migrations/9191_string_nonlocalized.py

```diff
@@ -0,0 +1,62 @@
+from django.db import migrations, models
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('models', '10515_i18n_plugins'),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql = """
+            INSERT INTO d_data_types(
+                datatype, 
+                iconclass, 
+                modulename,
+                classname, 
+                defaultconfig, 
+                configcomponent,
+                configname, 
+                isgeometric, 
+                defaultwidget,
+                issearchable
+            ) VALUES (
+                'non-localized-string',
+                'fa fa-file-code-o',
+                'datatypes.py',
+                'NonLocalizedStringDataType',
+                null,
+                'views/components/datatypes/non-localized-string',
+                'non-localized-string-datatype-config',
+                FALSE,
+                '46ef064b-2611-4708-9f52-60136bd8a65b',
+                TRUE
+            );
+            INSERT INTO widgets(
+                widgetid,
+                name,
+                component,
+                datatype,
+                defaultconfig
+            ) VALUES (
+                '46ef064b-2611-4708-9f52-60136bd8a65b',
+                'non-localized-text-widget',
+                'views/components/widgets/non-localized-text',
+                'non-localized-string',
+                '{
+                    "width": "100%",
+                    "maxLength": null,
+                    "uneditable": false,
+                    "placeholder": "Enter text",
+                    "defaultValue": ""
+                }'
+            );
+            """,
+            reverse_sql = """
+            DELETE FROM d_data_types
+                WHERE datatype = 'non-localized-string';
+            DELETE from widgets
+                WHERE widgetid = '46ef064b-2611-4708-9f52-60136bd8a65b';
+            """
```

Comment:
Although I haven't tested this explicitly I'm leery of deleting items we didn't explicitly add.  For example deleting a node/edge in a graph just because it references a datatype that now doesn't exist.  I'm pretty sure that's going to blow up the graph and at least will orphan any child nodes of the deleted node.  That seems unexpected.  If I was a graph designer I would be pretty bummed, and there would be no way to recover from that without going to the db and even then it would be tough to do.  It would be nice if instead we provide a sensible default datatype for those nodes (maybe semantic).  Or maybe leave it as-is.   I'm going to test this first though.

Reply:
I like the idea of a fallback. I'm sure you've noticed this already, but the reason Cyrus was deleting nodes was because the reverse migration failed to delete the datatype [like this](https://github.com/archesproject/arches/pull/10572#discussion_r1482201070)

Reply:
I tired adding the code that @chiatt  created above and it made my whole graph unavailable.  I then thought about changing the datatype to a localized string, but that also has substantial pitfalls (not the least of which is the ES mappings). 
The more I think about this the more I think that we shouldn't have a reverse migration at all.  That we should essentially leave the "new" datatype in place, even if a user wants to reverse the migration.  For one thing it won't hurt to leave it in.  For another, this is a new feature that we want users to have.  It causes a bunch of chaos if a user has started to use it and then for whatever reason wants to undo the migration.  Leaving it in place allows a user to manually undo the work **if they want to**, creating a migration path for their own data if they want by using existing methods built in to Arches.  
In this case at least, if we could re-imagine our whole "upgrade" to the localized  string datatype, we would have left in a non-localized version.